src/main/scripts/parse_synonyms.py

```diff
@@ -0,0 +1,187 @@
+    `parents`, then recursively calls itself to add `label` to any
+    grandparents. `childDepth` is decreased by 1 for each generation to prevent
+    exponentially large injections.
+
+    Parameters
+    ----------
+    relationships: Dict[str, Dict[str, List[str]]]
+        Maps terms to an inner dictionary containing arrays for "alternatives",
+        "parents", and "children".
+    label: str
+        The term to be added to its `parents`.
+    parents: List[str]
+        The direct parents of the current `label`
+    childDepth: int
+        The number of generations of children to inject for each term.
+        For example, a value of 2 would inject children and their children.
+        0 will only add alternative terms. Negative integers will add all
+        children, grandchildren, etc. Note that this may result in an
+        exponentially large number of terms
+    """
+    if childDepth != 0:
+        for parent in parents:
+            try:
+                relationships[parent]["children"].append(label)
+                # If the parent is equivalent to anything, also add label as a
+                # child of the equivalentParent
+                for equivalentParent in relationships[parent]["equivalent"]:
+                    relationships[equivalentParent]["children"].append(label)
+                addToParents(
+                    relationships,
+                    label,
+                    relationships[parent]["parents"],
+                    childDepth - 1,
+                )
+            except KeyError:
+                pass
+
+
+def main(inputFile: str, outputFile: str, mode: str, maxChildDepth: int):
+    """
+    Reads an CSV file of terminology and writes it into Solr synonym format
+    for use in synonym injection. Alternative terms are always written, and the
+    number of child terms is configurable by `maxChildDepth`.
+
+    Parameters
+    ----------
+    inputFile: str
+        CSV file to read ontology from.
+    outputFile: str
+        Solr synonym output file.
+    mode: str
+        Python file mode (w, a, ...) to use when writing the output file.
+    maxChildDepth: int
+        The maximum number of generations of children to inject for each term.
+        For example, a value of 2 would inject children and their children.
+        0 will only add alternative terms. Negative integers will add all
+        children, grandchildren, etc. Note that this may result in an
+        exponentially large number of terms
+    """
+    altIndices = []
+    parentIndices = []
+    equivalentIndices = []
+    equivalentPairs = {}
+    relationships = {}
+    with open(inputFile) as f:
+        reader = csv.reader(f)
+
+        # Dynamically determine header positions
+        headers = next(reader)
+        for i, header in enumerate(headers):
+            if "Label" == header.strip():
+                labelIndex = i
+            # elif "Class Type" == header:
+            #     classIndex = i
```

Comment:
Yeah. If memory serves, this is for a niche case in the PaNET ontology that was only used once or twice. At some point I was probably debating whether to support it or not (in case more terms using it were added) so didn't delete it entirely, but at this point if I wanted to put it back in I'd need to write it from scratch anyway since I can't remember what it actually means anymore.